docs/architecture/joint-access-storage-driver.md

```diff
@@ -0,0 +1,209 @@
+
+## Advanced Features
+
+Depending on the capabilities of the underlying file system, the Joint Access Storage Driver can benefit from more advanced functionality described here.
+
+### Versioning
+
+If the underlying file system is able to create versions of single resources (imagine a git based file system) this functionality could directly be used by Infinite Scale.
+
+In the current state of the Joint Access Storage Driver, versioning is not supported.
+
+### Trashbin
+
+If the underlying file system handles deleted files in a trash bin that allows restoring of previously removed files, this functionality could directly be used by Infinite Scale.
+
+If not available it will follow the [the Free Desktop Trash specificaton](https://specifications.freedesktop.org/trash-spec/trashspec-latest.html).
+
+In the current state of the Joint Access Storage Driver, trash bin is not supported.
+
+## Limitations
+
+As of Q2/2024 the Joint Access Storage Driver is not officially supported and in technical preview state.
+
+The tech preview comes with the following limitations:
+
+1. Only inotify and GPFS file system change notification methods are supported
+1. Advanced features versioning and trashbin are not supported yet
+1. The space/project folders in the filesystem are named after the UUID, not the real space name
+1. No CephFS support yet
+1. Postprocessing (ie. anti virus check) does not happen for file actions outside of Infinite Scale
+
+## Setup
+
+This describes the steps to use the Joint Access Storage Driver storage driver with Infinite Scale.
+
+It is possible to use different storage drivers in the same Infinite Scale installation. For example it is possible to set up one space running on Joint Access Storage Driver while others run Decomposed FS.
+
+### Prerequisites
+
+To use the Joint Access Storage Driver, the following prerequisites have to be fulfilled:
+
+1. There must be storage available to store meta data and blobs, available under a root path
+1. When using inotify, the storage must be local on the same machine. Network mounts do not work with inotify. `inotifywait` needs to be installed.
+1. The storage root path must be writeable and executable by the same user Infinite Scale is running under
+1. An appropiate version of Infinite Scale is installed, version number 5.0.5 and later
+1. Nats-js-kv as cache service
+
+
+### Setup Configuration
+
+This is an example configuration with environment variables that configures Infinite Scale to use Joint Access Storage Driver for all spaces it works with, ie. Personal and Project Spaces:
+
+```
+export STORAGE_USERS_DRIVER="posix"
+export STORAGE_USERS_POSIX_ROOT="/home/kf/tmp/posix-storage"
+export STORAGE_USERS_POSIX_WATCH_TYPE="inotifywait"
+export STORAGE_USERS_ID_CACHE_STORE="nats-js-kv"
+export STORAGE_USERS_ID_CACHE_STORE_NODES="localhost:9233"
```

Comment:
Would'nt it be good, to be inline with the name of the driver to have all envvars respectively key-values use the driver name? Like `STORAGE_USERS_POSIX_ROOT` --> `STORAGE_USERS_JASD_ROOT` This would directly match its purpose and would match the name when documenting. Readers immediately have an identification. `Joint Access Storage Driver` and `POSIX` do not match when reading...

Reply:
Yes, I agree. That requires code changes, and once these are done, we will bring the changes to this doc in the same PR.